### PR TITLE
Fix issues with LazyFiles

### PR DESCRIPTION
**Pull request by fzimmermann89**
The closure change is a straight forward change like in https://github.com/kripken/emscripten/blob/master/src/library_fs.js#L200

The solution to the gzip issue is not pretty, but works. I can't write a test for it, though (needs gzip server)

### CODE REVIEW

**Comment by fzimmermann89** [comment]
google's content compression seems to remove the content-datalength header on some files. The last commit uses the work-around in these cases, too.

File: src/library_fs.js [hunk 1]
```diff
@@ -1596,6 +1596,8 @@ mergeInto(LibraryManager.library, {
         var datalength = Number(xhr.getResponseHeader("Content-length"));
         var header;
         var hasByteServing = (header = xhr.getResponseHeader("Accept-Ranges")) && header === "bytes";
+        var usesGzip = (header = xhr.getResponseHeader("Content-Encoding"))&& header === "gzip";
```
Comment (kripken) [inline src/library_fs.js]: space before `&&`

File: src/library_fs.js [hunk 2]
```diff
@@ -1640,6 +1641,15 @@ mergeInto(LibraryManager.library, {
           return lazyArray.chunks[chunkNum];
         });
 
+
+        if (usesGzip||!datalength){
```
Comment (kripken) [inline src/library_fs.js]: spaces around `||` and after `)`

File: src/library_fs.js [hunk 3]
```diff
@@ -1640,6 +1641,15 @@ mergeInto(LibraryManager.library, {
...
+            //if the server uses gzip or doesn't supply the length, we have to download the whole file to get the (uncompressed) length
+            chunkSize = datalength = 1; //this will force getter(0)/doXHR do download the whole file
+            datalength=this.getter(0).length;
```
Comment (kripken) [inline src/library_fs.js]: spaces around `=`

File: src/library_fs.js [hunk 4]
```diff
@@ -1640,6 +1641,15 @@ mergeInto(LibraryManager.library, {
           return lazyArray.chunks[chunkNum];
         });
 
+
+        if (usesGzip||!datalength){
+            //if the server uses gzip or doesn't supply the length, we have to download the whole file to get the (uncompressed) length
```
Comment (kripken) [inline src/library_fs.js]: space after //`, and indentation should be 2 spaces (i see some of the code you are replacing had wrong formatting, let's fix this while you replace the code)

**Comment by fzimmermann89** [comment]
Sry, I hope I fixed all the issues.

**Comment by kripken** [comment]
Looks good. Please just squash the commits.

**Comment by fzimmermann89** [comment]
Ok?

**Comment by kripken** [comment]
Great, thanks!

**Pull request merged**